contessa/consistency_checker.py

```diff
@@ -0,0 +1,136 @@
+        right_check_table: Dict,
+        result_table: Dict,
+        context: Optional[Dict] = None,
+    ):
+        left_check_table = Table(**left_check_table)
+        right_check_table = Table(**right_check_table)
+        result_table = ResultTable(**result_table)
+        context = self.get_context(left_check_table, right_check_table, context)
+
+        result = self.do_consistency_check(
+            check_type, left_check_table, right_check_table, context
+        )
+
+        quality_check_class = create_default_check_class(
+            result_table, check_type="consistency"
+        )
+        self.ensure_table(quality_check_class)
+        self.insert(quality_check_class, result)
+
+    @staticmethod
+    def get_context(
+        left_check_table: Table,
+        right_check_table: Table,
+        context: Optional[Dict] = None,
+    ) -> Dict:
+        """
+        Construct context to pass to executors. User context overrides defaults.
+        """
+        ctx_defaults = {
+            "left_table_fullname": left_check_table.fullname,
+            "right_table_fullname": right_check_table.fullname,
+            "task_ts": datetime.now(),
+        }
+        ctx_defaults.update(context)
+        return ctx_defaults
+
+    def do_consistency_check(
+        self,
+        check_type: str,
+        left_check_table: Table,
+        right_check_table: Table,
+        context: Dict = None,
+    ):
+        """
+        Run quality check for all rules. Use `qc_cls` to construct objects that will be inserted
+        afterwards.
+        """
+        left_result = self.run_query(
+            left_check_table.fullname, self.left_conn, check_type
+        )
+        right_result = self.run_query(
+            right_check_table.fullname, self.right_conn, check_type
+        )
+        return {
+            "check": {"name": check_type, "description": ""},
+            "status": left_result == right_result,
+            "left_table_name": left_check_table.fullname,
+            "right_table_name": right_check_table.fullname,
+            "context": context,
+        }
+
+    def run_query(self, table_name: str, conn: Connector, type: str):
+        if type == self.COUNT:
+            column = "count(*)"
+        else:
+            column = "*"
+        query = f"""
+            SELECT { column }
+            FROM { table_name }
+        """
+        result = [r for r in conn.get_records(query)]
+        return result
+
+    def ensure_table(self, qc_cls):
+        """
+        Create table for QualityCheck class if it doesn't exists. E.g. quality_check_
+        """
+        try:
+            qc_cls.__table__.create(bind=self.right_conn.engine)
+            logging.info(f"Created table {qc_cls.__tablename__}.")
+        except sqlalchemy.exc.ProgrammingError:
+            logging.info(
+                f"Table {qc_cls.__tablename__} already exists. Skipping creation."
+            )
+
+    def insert(self, dc_cls, result):
```

Comment:
maybe this could be extracted in some clever way, considerring the existing runner?

Reply:
I meant some session generator as context manager. example stolen from Airflow, i'd do it with use of class context manager not the function on the Connector class. It could hide the session management. I believe i should be more explicit next time. Let's refactor it later :) 

```python
@contextlib.contextmanager
def create_session():
    """
    Contextmanager that will create and teardown a session.
    """
    session = settings.Session()
    try:
        yield session
        session.commit()
    except Exception:
        session.rollback()
        raise
    finally:
        session.close()
```